### DISCUSSION TITLE
Why can an LLM tell the date correctly?

### DISCUSSION THREAD

**Discussion by CLRafaelR** [mckaywrigley/chatbot-ui#1688] [Help]
![image](https://github.com/mckaywrigley/chatbot-ui/assets/42314553/a9733c76-b4cb-44f9-b2ef-da7d4c1734a7)

As far as I understand, OpenAI's GPT models themselves should not have access to date information. However, as shown in the attached screenshot, when I ask gpt-4 vision for the date via Chatbot-UI, it gives me the correct date for today, although it fails to answer the day of the week (Monday is the correct answer). Is there any implementation/mechanism on the Chatbot-UI that allows the model to get the server's date information?

**Comment by secondtruth** [comment]
> Is there any implementation/mechanism on the Chatbot-UI that allows the model to get the server's date information?

Yes, ChatbotUI's full system prompt contains the current date.

https://github.com/mckaywrigley/chatbot-ui/blob/a3364dba7b04b0197e1783ca0bae1669705e61a2/lib/build-prompt.ts#L17

**Reply by CLRafaelR** [reply]
@secondtruth , I apologise for the delayed response and appreciate your patience. Thank you for enlightening me on how Chatbot-UI communicates date information to an LLM. Additionally, I'm grateful for your effort in directing me to the specific code segment that furnishes the LLM with date details.
